OPAL/br/src/main/scala/org/opalj/br/analyses/cg/EntryPointFinder.scala

```diff
@@ -357,36 +357,71 @@ object AllEntryPointsFinder extends EntryPointFinder {
  */
 object AndroidEntryPointsFinder extends EntryPointFinder {
 
-    val activityEPS: List[String] = List("onCreate", "onRestart", "onStart", "onResume",
-        "onStop", "onDestroy", "onActivityResult")
-    val serviceEPS: List[String] = List("onCreate", "onStartCommand", "onBind", "onStart")
-    val contentProviderEPS: List[String] = List("onCreate", "query", "insert", "update")
-    val locationListenerEPS: List[String] = List("onLocationChanged", "onProviderDisabled", "onProviderEnabled",
-        "onStatusChanged")
-    val onNmeaMessageListenerEPS: List[String] = List("onNmeaMessage")
-    val broadcastReceiverEPS: List[String] = List("onReceive")
-    val defaultEPS = Map("android/app/Activity" -> activityEPS, "android/app/Service" -> serviceEPS,
-        "android/content/ContentProvider" -> contentProviderEPS,
-        "android/location/LocationListener" -> locationListenerEPS,
-        "android/location/onNmeaMessageListener" -> onNmeaMessageListenerEPS,
-        "android/content/BroadcastReceiver" -> broadcastReceiverEPS)
+    val activityEntryPoints: List[(String, MethodDescriptor)] = List(
+        ("onCreate", MethodDescriptor.apply(FieldType("Landroid/os/Bundle;"), VoidType)),
+        ("onRestart", MethodDescriptor.NoArgsAndReturnVoid),
+        ("onStart", MethodDescriptor.NoArgsAndReturnVoid),
+        ("onResume", MethodDescriptor.NoArgsAndReturnVoid),
+        ("onStop", MethodDescriptor.NoArgsAndReturnVoid),
+        ("onDestroy", MethodDescriptor.NoArgsAndReturnVoid),
+        ("onActivityResult", MethodDescriptor.NoArgsAndReturnVoid)
+    )
+    val serviceEntryPoints: List[(String, MethodDescriptor)] = List(
+        ("onCreate", MethodDescriptor.NoArgsAndReturnVoid),
+        ("onStartCommand", MethodDescriptor.apply(FieldTypes(FieldType("Landroid/content/Intent;"), IntegerType, IntegerType), VoidType)),
+        ("onBind", MethodDescriptor.apply(FieldType("Landroid/content/Intent;"), VoidType)),
+        ("onRebind", MethodDescriptor.apply(FieldType("Landroid/content/Intent;"), VoidType)),
+        ("onStart", MethodDescriptor.apply(FieldTypes(FieldType("Landroid/content/Intent;"), IntegerType), VoidType)),
+        ("onDestroy", MethodDescriptor.NoArgsAndReturnVoid)
+    )
+    val contentProviderEntryPoints: List[(String, MethodDescriptor)] = List(
+        ("onCreate", MethodDescriptor.NoArgsAndReturnVoid),
+        ("query", null),
```

Comment:
What are these null descriptors? Are these methods where multiple descriptors apply? If it is only two or three, maybe still just list them all. If you need to match all descriptors, I'd prefer to change the type here to Option[MethodDescriptor] and have None instead of null for more idiomatic Scala.